### PR TITLE
Move src/platform/tests to auto-generation of unit test drivers

### PR DESCRIPTION
**Pull request by andy31415**
Part of a larger cleanup. Replacing folder one by one since mass-replacement seemed to have resulted in some harder to track dependency errors.

### CODE REVIEW

**Comment by andy31415** [comment]
It seems that PlatformMgr tests used NOT to run on zephyr previously (due to CHIP_REGISTER_TEST_SUITE not being used).

new code uses it, so we have to disable the test from the build. Apparently running the system event loop crashes (guessing there is an event loop already that runs all the tests, hence the failure).

I manually disabled the test. Unsure how to run it locally.

File: src/platform/tests/BUILD.gn [hunk 1]
```diff
@@ -23,25 +23,17 @@ if (chip_device_platform != "none") {
...
-      "TestPlatformMgr.h",
-      "TestPlatformTime.cpp",
-      "TestPlatformTime.h",
-    ]
+    test_sources = []
 
-    tests = [ "TestPlatformMgr" ]
+    if (current_os != "zephyr") {
+      # TODO: unclear as to why this fails on zephyr
+      #       pre-conversion this test was not registered as CHIP_REGISTER_TEST_SUITE
```
Comment (vivien-apple) [inline src/platform/tests/BUILD.gn:30]: Can you open a followup so it can be tracked down ?

File: src/platform/tests/TestCHIPoBLEStackMgr.cpp [hunk 2]
```diff
@@ -36,6 +37,8 @@ void EventHandler(const chip::DeviceLayer::ChipDeviceEvent * event, intptr_t arg
 
 int TestCHIPoBLEStackManager()
 {
+    chip::Platform::MemoryInit();
```
Comment (vivien-apple) [inline src/platform/tests/TestCHIPoBLEStackMgr.cpp:40]: I wish this does not need to be added to all tests manually.

**Review by vivien-apple — APPROVED** [review]

**Review by rwalker-apple — APPROVED** [review]

**Review by mspang — APPROVED** [review]

File: src/platform/tests/TestCHIPoBLEStackMgr.cpp (hunk 2 above)
Comment (andy31415) [reply src/platform/tests/TestCHIPoBLEStackMgr.cpp:40]: I believe it depends if memory is used or not. I blindly copied stuff that was extra in 'Driver' into the test code.

**Review by BroderickCarlin — APPROVED** [review]

**Pull request merged**